portal/src/main/webapp/js/src/study-view/data/StudyViewProxy.js

```diff
@@ -460,6 +573,8 @@ var StudyViewProxy = (function() {
         getArrDataBySampleIds: getArrDataBySampleIds,
         getAttrData: function(){ return obtainDataObject.attr;},
         getMutatedGenesData: getMutatedGenesData,
+        getMutatedGeneDataBasedOnSampleIds: getMutatedGeneDataBasedOnSampleIds,
```

Comment:
General comment: 
I would favor StudyViewProxy connecting in some way to the new API exposed in cbioportal-client.js and cbioportal-datamanager.js (also known as QuerySession)

These methods like StudyViewProxy.getMutatedGeneDataBasedOnSampleIds seem to repeat things that are (or should be) also available in QuerySession API. 

@adamabeshouse, @jjgao : what do you think?